laravel/app/Http/Controllers/FollowController.php

```diff
@@ -34,14 +34,21 @@ public function create()
      * @param  \Illuminate\Http\Request  $request
      * @return \Illuminate\Http\Response
      */
-    public function store(Request $request)
-    {
-        $follow = new Follow;
-        $follow->user_id = Auth::id();
-        $follow->follow_user_id = $request->follow_user_id;
-        $follow->save();
-        return redirect()->back();
-    }
+     public function store(Request $request)
+     {
+         if ($request->follow_user_id == Auth::id())
+         {
+             return back()->with('error', '自分自身をフォローすることはできません');
+         }else{
+             $follow = new Follow;
+             //フォローする側
+             $follow->user_id = Auth::id();
+             //フォローされる側
+             $follow->follow_user_id = $request->follow_user_id;
+             $follow->save();
+             return redirect()->back();
+         }
```

Comment:
この様なケースの場合は、ifの条件に合致したらreturnしているので elseは書かない方が階層が浅くなるのでおすすめです。
```suggestion
         if ($request->follow_user_id == Auth::id())
         {
             return back()->with('error', '自分自身をフォローすることはできません');
         }
         $follow = new Follow;
         //フォローする側
         $follow->user_id = Auth::id();
         //フォローされる側
         $follow->follow_user_id = $request->follow_user_id;
         $follow->save();
         return redirect()->back();
```